src/AppBundle/Issues/IssueListener.php

```diff
@@ -58,14 +58,39 @@ public function handleCommentAddedEvent($issueNumber, $comment)
      * Adds a "Needs Review" label to new PRs.
      *
      * @param int $prNumber The number of the PR
+     * @param int $prTitle  The title of the PR
+     * @param int $prBody   The full text description of the PR
      *
      * @return string The new status
      */
-    public function handlePullRequestCreatedEvent($prNumber)
+    public function handlePullRequestCreatedEvent($prNumber, $prTitle, $prBody)
     {
+        $prLabels = array();
+
+        // new PRs always require review
         $newStatus = Status::NEEDS_REVIEW;
+        $prLabels[] = $newStatus;
+
+        // the PR title usually indicates the affected component
```

Comment:
Assuming the first bracket will be the component name is far from true. Usually, `[WIP]`, `[RFC]`, `[POC]` are put before the change. Even more, some people tend to include the version number in the PR title, also within brackets. This is placed before the component brackets as well.

Imo, we should get a list of available labels for a repository and check all brackets as long as the value inside the brackets do not match a label